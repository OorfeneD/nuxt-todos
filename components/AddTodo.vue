<template>
  <div class="flex w-full justify-between align-middle">
    <input v-model="title" class="focus:shadow-inner transition shadow-xs focus:outline-none border border-gray-200 text-gray-600 px-3 py-2 rounded-md flex-grow mr-2" type="text" placeholder="todo title ...">
    <button class="hover:bg-indigo-500 text-gray-600 hover:text-white transition px-2 py-2 border border-gray-200 rounded-md hover:shadow-sm" @click="$emit('addtodo', { content: { title }})">
      Add
    </button>
  </div>
</template>

<script>
export default {
  data () {
    return {
      title: ''
    }
  }
}
</script>

<style>

</style>
