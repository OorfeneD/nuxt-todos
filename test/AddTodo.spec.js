import { shallowMount } from '@vue/test-utils'
import AddTodo from '../components/AddTodo.vue'

describe('AddTodo Component', () => {
  const wrapper = shallowMount(AddTodo)
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })
  test('instance contains `title` empty string variable', () => {
    expect(wrapper.vm).toStrictEqual(expect.objectContaining({ title: '' }))
  })
  test('contains AddTodo html', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('emits event on form submit', () => {
    wrapper.find('input').setValue('test')
    wrapper.find('form').trigger('submit')
    expect(wrapper.emitted().addtodo).toBeTruthy()
  })
  test('input value was cleared after submit', () => {
    expect(wrapper.vm.title.length).toBe(0)
  })
  test('emitted event contains `test` string', () => {
    expect(wrapper.emitted('addtodo')[0]).not.toBeUndefined()
    expect(wrapper.emitted('addtodo')[0][0]).toStrictEqual(
      expect.objectContaining(
        { content: { title: 'test' } }
      )
    )
  })
})
