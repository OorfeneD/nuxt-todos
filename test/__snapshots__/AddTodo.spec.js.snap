// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo Component contains AddTodo html 1`] = `
"<form class=\\"flex w-full justify-between align-middle\\"><input type=\\"text\\" placeholder=\\"todo title ...\\" required=\\"required\\" class=\\"focus:shadow-inner transition shadow-xs focus:outline-none border border-gray-200 text-gray-600 px-3 py-2 rounded-md flex-grow mr-2\\"> <button class=\\"hover:bg-indigo-500 text-gray-600 hover:text-white transition px-2 py-2 border border-gray-200 rounded-md hover:shadow-sm\\">
    Add
  </button></form>"
`;
